### ISSUE TITLE
Multiprocessing Error on Windows

### ISSUE THREAD

**Issue by Giszy** [pytorch/pytorch#4098]
@peterjc123, @yf225
When we call 
```python
Data.DataLoader(dataset=torch_dataset, batch_size=BATCH_SIZE, shuffle=True, num_workers=2,)
```
it will show errors:
Traceback (most recent call last):
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <module>
  File "C:\C:\Python36_X64\lib\multiprocessing\forking.py", line 381, in main
  File "C:\C:\Python36_X64\lib\multiprocessing\forking.py", line 381, in main
    self = load(from_parent)
  File "C:\C:\Python36_X64\lib\pickle.py", line 1384, in load
    self = load(from_parent)
  File "C:\C:\Python36_X64\lib\pickle.py", line 1384, in load
    return Unpickler(file).load()
      File "C:\C:\Python36_X64\lib\pickle.py", line 864, in load
return Unpickler(file).load()
  File "C:\C:\Python36_X64\lib\pickle.py", line 864, in load
    dispatch[key](self)
  File "C:\C:\Python36_X64\lib\pickle.py", line 1221, in load_build
        setstate = getattr(inst, "__setstate__", None)
dispatch[key](self)
  File "C:\C:\Python36_X64\lib\site-packages\torch\multiprocessing\queue.py", line 27, in __getattr__
  File "C:\C:\Python36_X64\lib\pickle.py", line 1221, in load_build
    return getattr(self.conn, name)
  File "C:\C:\Python36_X64\lib\site-packages\torch\multiprocessing\queue.py", line 27, in __getattr__
    return getattr(self.conn, name)
  File "C:\C:\Python36_X64\lib\site-packages\torch\multiprocessing\queue.py", line 27, in __getattr__
    return getattr(self.conn, name)
      File "C:\C:\Python36_X64\lib\site-packages\torch\multiprocessing\queue.py", line 27, in __getattr__
...
    return getattr(self.conn, name)
RuntimeError: maximum recursion depth exceeded

if we set num_workers=2, the code can run.

**Comment by peterjc123** [comment]
@Giszy Don't expect PyTorch to work under Python 2.7. The serialization part is broken and there's no possible fix for that.

**Comment by Giszy** [comment]
1. PyTorch can work under python 2.7 on windows.
2. I have fixed the errors. It is necessary to modify "torch\multiprocessing\queue.py".

**Comment by peterjc123** [comment]
@Giszy Yes, you may be right. But there're some major factors that I don't want to support this.
1. The serialization part is broken. You can just try `torch.save` or `torch.load`, and you will get a error. 
2. CUDA is not supported. No compiler is usable to build that under Windows.
3. Python 2 only supports the integral type `long` as the argument type in the Python function, which is not so friendly under Windows.
The second reason is critical. It means that the users using it will be very limited. They may just build from source when needed.

**Comment by Giszy** [comment]
@peterjc123 
1. You are right. but the multiprocessing issue is the same as in python 3.x
2. python2.7 + vc2015 can solve this problem
3. ...

**Comment by peterjc123** [comment]
@Gizzy How can you build Python 2.7 extensions with VS2015?

**Comment by Giszy** [comment]
open the file
"C:\Python27_x64\Lib\distutils\msvc9compiler.py" and locate line 244, replace "toolskey = "VS%0.f0COMNTOOLS" % version" with "toolskey = "VS140COMNTOOLS""

**Comment by peterjc123** [comment]
@Giszy Since you found out the fix for that, you can just send a PR. But I guess there won't have support for Python 2.7, as this method is not well documented and supported by the official Python team.

**Comment by Giszy** [comment]
I compiled it because of a package can run only under python 2.7.  But the problem here has no relation with python 2.7. In python 3.x, the issue is the same.

**Comment by peterjc123** [comment]
@Giszy So would you please raise a PR or share with us your fix so that we can merge it into the master branch?

**Comment by Giszy** [comment]
How raise a PR?

**Comment by peterjc123** [comment]
@Giszy First, you should fork this repo. And then, if there're only changes with a file, you can edit it online. Otherwise, you should make a commit and push it. Finally, you visit the main page of this repo and you can see compare & pull request, click the button and you can finish with that.

**Comment by peterjc123** [comment]
@Giszy Maybe you should close this issue, as this has been addressed by #4226.
